tests/unit/backend/services/webhook.service.spec.ts

```diff
@@ -0,0 +1,151 @@
+import axios from 'axios'
+import { ObjectID } from 'bson'
+import mongoose from 'mongoose'
+
+import { getEncryptSubmissionModel } from 'src/app/models/submission.server.model'
+import {
+  handleWebhookFailure,
+  handleWebhookSuccess,
+  postWebhook,
+} from 'src/app/services/webhooks.service'
+import { AxiosError, WebhookValidationError } from 'src/app/utils/custom-errors'
+
+import dbHandler from '../helpers/jest-db'
+
+const EncryptSubmission = getEncryptSubmissionModel(mongoose)
+
+describe('WebhooksService', () => {
+  beforeAll(async () => await dbHandler.connect())
+  afterEach(async () => await dbHandler.clearDatabase())
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  let testEncryptForm
+  let testEncryptSubmission
+  let testWebhookParam
+  let testConfig
+  let testSubmissionWebhookView
+  const MOCK_ADMIN_OBJ_ID = new ObjectID()
+  const MOCK_SIGNATURE = 'mock-signature'
+  const MOCK_WEBHOOK_URL = '/webhook-endpoint'
+  const MOCK_NOW = Date.now()
+  const ERROR_MSG = 'test-message'
+  const MOCK_SUCCESS_RESPONSE = {
+    data: {
+      result: 'test-result',
+    },
+    status: 200,
+    statusText: 'success',
+    headers: {},
+    config: {},
+  }
+  const MOCK_STRINGIFIED_PARSED_RESPONSE = {
+    data: '{"result":"test-result"}',
+    status: 200,
+    statusText: 'success',
+    headers: '{}',
+  }
+
+  beforeEach(async () => {
+    const preloaded = await dbHandler.insertFormCollectionReqs({
+      userId: MOCK_ADMIN_OBJ_ID,
+    })
+    testEncryptForm = preloaded.encryptForm
+    testEncryptSubmission = preloaded.encryptSubmission
+    testSubmissionWebhookView = testEncryptSubmission.getWebhookView()
+    testWebhookParam = {
+      webhookUrl: MOCK_WEBHOOK_URL,
+      submissionWebhookView: testSubmissionWebhookView,
+      submissionId: testEncryptSubmission._id,
+      formId: testEncryptForm._id,
+      now: MOCK_NOW,
+      signature: MOCK_SIGNATURE,
+    }
+    testConfig = {
+      headers: {
+        'X-FormSG-Signature': `t=${MOCK_NOW},s=${testEncryptSubmission._id},f=${testEncryptForm._id},v1=${MOCK_SIGNATURE}`,
+      },
+      maxRedirects: 0,
+    }
+  })
+
+  function delay(delayInms) {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve()
+      }, delayInms)
+    })
+  }
+
+  describe('handleWebhookFailure', () => {
+    it('should update submission document with failed webhook response', async () => {
+      // Act
+      const error = new AxiosError(ERROR_MSG, MOCK_SUCCESS_RESPONSE)
+      handleWebhookFailure(error, testWebhookParam)
+      await delay(3000)
```

Comment:
Can we not convert handleWebhookFailure to return a promise instead of void? Can always convert it to return  `Promise.allSettled`, then we can `await` instead of doing the delay.

Reply:
will do